expression/integration_test.go

```diff
@@ -1098,6 +1098,17 @@ func (s *testIntegrationSuite) TestTimeBuiltin(c *C) {
 	result = tk.MustQuery("select to_seconds('0000-00-00');")
 	result.Check(testkit.Rows("<nil>"))
 
+	result = tk.MustQuery("select timestamp('2003-12-31'), timestamp('2003-12-31 12:00:00','12:00:00');")
+	result.Check(testkit.Rows("2003-12-31 00:00:00 2004-01-01 00:00:00"))
+	result = tk.MustQuery("select timestamp(20170118123950.123), timestamp(20170118123950.999);")
+	result.Check(testkit.Rows("2017-01-18 12:39:50.123 2017-01-18 12:39:50.999"))
+	result = tk.MustQuery("select timestamp('2003-12-31', '01:01:01.01'), timestamp('2003-12-31.1234', '01:01:01.01');")
+	result.Check(testkit.Rows("2003-12-31 01:01:01.01 2003-12-31 01:01:01.1334"))
+	result = tk.MustQuery("select timestamp('2003-12-31', 1), timestamp('2003-12-31', -1);")
+	result.Check(testkit.Rows("2003-12-31 00:00:01 2003-12-30 23:59:59"))
+	result = tk.MustQuery("select timestamp('2003-12-32', '2000-12-12 01:01:01.01');")
```

Comment:
add test for invalid input and zero time

Reply:
done